source/http_message.c

```diff
@@ -0,0 +1,468 @@
+/*
+ * Copyright 2010-2018 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License").
+ * You may not use this file except in compliance with the License.
+ * A copy of the License is located at
+ *
+ *  http://aws.amazon.com/apache2.0
+ *
+ * or in the "license" file accompanying this file. This file is distributed
+ * on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
+ * express or implied. See the License for the specific language governing
+ * permissions and limitations under the License.
+ */
+
+#include "module.h"
+
+#include "class_binder.h"
+#include "http_message.h"
+
+#include <aws/http/request_response.h>
+
+static struct aws_napi_class_info s_request_clazz;
+
+static aws_napi_method_fn s_request_constructor;
+
+static aws_napi_property_get_fn s_method_get;
+static aws_napi_property_set_fn s_method_set;
+static aws_napi_property_get_fn s_path_get;
+static aws_napi_property_set_fn s_path_set;
+static aws_napi_property_set_fn s_body_set;
+static aws_napi_property_get_fn s_num_headers_get;
+
+static aws_napi_method_fn s_add_header;
+static aws_napi_method_fn s_set_header;
+static aws_napi_method_fn s_get_header;
+static aws_napi_method_fn s_erase_header;
+
+napi_status aws_napi_http_message_bind(napi_env env, napi_value exports) {
+
+    static const struct aws_napi_method_info s_request_constructor_info = {
+        .name = "HttpRequest",
+        .method = s_request_constructor,
+
+        .num_arguments = 0,
+        .arg_types = {napi_string, napi_string, napi_external, napi_undefined},
+    };
+
+    static const struct aws_napi_property_info s_request_properties[] = {
+        {
+            .name = "method",
+            .type = napi_string,
+            .getter = s_method_get,
+            .setter = s_method_set,
+            .attributes = napi_enumerable | napi_writable,
+        },
+        {
+            .name = "path",
+            .type = napi_string,
+            .getter = s_path_get,
+            .setter = s_path_set,
+            .attributes = napi_enumerable | napi_writable,
+        },
+        {
+            .name = "body",
+            .setter = s_body_set,
+            .attributes = napi_enumerable | napi_writable,
+        },
+        {
+            .name = "num_headers",
```

Comment:
Should this be like ```.size``` or ```.length```?

Reply:
Will revisit when Mike finishes his Headers class, but for now, request.length isn't really clear enough.